include/cblox/core/tsdf_submap_collection.h

```diff
@@ -37,6 +37,10 @@ class TsdfSubmapCollection {
 
   // Creates a new submap on the top of the collection
   void createNewSubMap(const Transformation &T_M_S, SubmapID submap_id);
+  void createNewSubMap(const Transformation &T_M_S);
+
+  // Create a new submap which duplicates an existing source submap
+  bool duplicateSubMap(const cblox::SubmapID source_submap_id, const cblox::SubmapID new_submap_id);
```

Comment:
I think this method would be better as a member of the submap class which implements a "deep copy". Something like:

`submap_2 = submap_1.copy()`

It seems strage to me that the collection would take responsibility for copying submap. Deep copy operations are fairly common.